BackEndFlask/models/utility.py

```diff
@@ -14,3 +22,49 @@ def get_courses_by_user_courses_by_user_id(user_id):
     except SQLAlchemyError as e:
         error = str(e.__dict__['orig'])
         return error
+
+def get_users_by_course_id_and_role_id(course_id, role_id):
```

Comment:
I am not a fan this function and the following functions. 

I don't understand the need to convert to a list of users rather than just using the SQLAlchemy object as we do throughout the model. If we got rid of this, we could simplify the function by just using query(User).

Reply:
In get_users_by_course_id_and_role_id(), I query the User and their corresponding role in the course. A tuple is returned of the SQLAlchemy User and the integer role_id. I was unable to figure out how to specify in the Marshmallow class to recognize the tuple of the SQLAlchemy User and the integer role_id to be parsed and places into the returned JSON. The logic that I have to manually parse the SQLAlchemy User and the corresponding role_id is the only working solution I was able to achieve.

Reply:
The same problem exists with get_users_by_role_id().

Reply:
I solved this issue in the RatingsRoutes branch. The returned Marshmallow returns data that is derived from multiple tables. Checkout Rating_routes.py.

Reply:
I have solved the issue by specifying what attributes of the Course and User table to pull to allow marshmallow to correctly parse the SQLAlchemy object and convert it to JSON!